doc/source/plugins.rst

```diff
@@ -0,0 +1,209 @@
+*******************
+The following is a summary of the functions available to a PluginInstance object
+
+Event Handlers
+==============
+
+* ``start``: Called when user “Activates” the plugin
+* ``update``: Called when when instance updates (multiple times per second)
+* ``on_run``: Called when user presses "Run"
+* ``on_stop``: Called when user disconnects or plugin crashes
+* ``on_advanced_settings``: Called when user presses "Advanced Settings"
+* ``on_complex_added``: Called whenever a complex is added to the workspace.
+* ``on_complex_removed``: Called whenever a complex is removed from the workspace.
+* ``on_presenter_changed``: Called when room's presenter changes.
+
+Spatial Actions
+===============
+
+* ``zoom_on_structures``: Repositions and resizes the workspace such that the provided structure(s) will be in the center of the users view.
+* ``center_on_structures``: Repositions the workspace such that the provided structure(s) will be in the center of the world.
+* ``request_presenter_info``: Requests presenter account info (unique ID, name, email)
+* ``request_controller_transforms``: Requests presenter controller info (head position, head rotation, left controller position, left controller rotation, right controller position, right controller rotation)
+
+IO/Streaming
+============
+
+* ``save_files``: Save files on the machine running Nanome, and returns result
+* ``create_writing_stream``: Create a stream allowing to continuously update properties of many objects
+* ``create_reading_stream``: Create a stream allowing to continuously receive properties of many objects
+* ``open_url``: Opens a URL in Nanome's computer browser
+* ``send_files_to_load``: Send file(s) to Nanome to load directly using Nanome's importers.
+* ``request_export``: Request a file export using Nanome exporters
+* ``set_plugin_list_button``: Set text and/or usable state of the buttons on the plugin connection menu in Nanome
+
+Workspace API Actions
+==========================
+
+* ``request_workspace``: Request the entire workspace, in deep mode
+* ``add_to_workspace``: Add a list of complexes to the current workspace
+* ``request_complex_list``: Request the list of all complexes in the workspace, in shallow mode
+* ``request_complexes``: Requests a list of complexes by their indices
+* ``update_workspace``: Replace the current workspace in the scene by the workspace in parameter
+* ``send_notification``: Send a notification to the user
+* ``update_structures_deep``: Update the specific molecular structures in the scene to match the structures in parameter.
+* ``update_structures_shallow``: Update the specific molecular structures in the scene to match the structures in parameter
+* ``apply_color_scheme``: Apply a color scheme to selected atoms.
+
+Menus/Stacks
+============
+
+* ``update_menu``: Update the menu in Nanome
+* ``update_content``: Update specific UI elements (button, slider, list...)
+* ``update_node``: Updates layout nodes and their children
```

Comment:
Swap position of `update_content` and `update_node`, then the ordering goes general to specific. (if you `update_node`, all child contents are updated too)